### PR TITLE
ecmascript 2021

### PR DESCRIPTION
**Pull request by soulmonk**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -116,7 +116,7 @@
 		"es6": true
 	},
 	"parserOptions": {
-		"ecmaVersion": 2018,
+		"ecmaVersion": 2021,
```
Comment (hilleer) [inline .eslintrc.json:119]: Hm... What about all the services services/modules not compatible with this? I notice `debitoor-app` is still running [on node 12](https://github.com/debitoor/debitoor-app/blob/master/package.json#L218) for example. Wouldn't that cause issues and make it incompatible?

Comment (soulmonk) [reply .eslintrc.json:119]: yep, indeed (
no I'm think to change only in `"ecmaVersion": 2021,` only api (

Comment (hilleer) [reply .eslintrc.json:119]: I have previously seen projects where they managed to create multiple configs in their eslint configuration. Would be nice to do something similar here, if possible :)

Comment (hilleer) [reply .eslintrc.json:119]: Accounting/integrations/platform has many services where I believe `2021` would make sense too.

Comment (soulmonk) [reply .eslintrc.json:119]: something like this https://github.com/debitoor/eslint-config-api ? )
quite strange to use https://github.com/debitoor/debitoor-api/blob/0f92f29f7aa3e1291e481ee1f7d0690d7ef11044/package.json#L166-L167

Comment (hilleer) [reply .eslintrc.json:119]: Hm, no. Not exactly. So you just have one package and on usage, when extending, you define a specific config, unless you are using default/latest. Imagine `.eslintrc.yml` files:

Default:

```yaml
extends: "@debitoor/eslint-config-debitoor"
```

Use "deprecated", old version, ecma specific version or node version :

```yaml
extends: "@debitoor/eslint-config-debitoor/es-2015"
```

I will try to remember where I saw it or if I can find a working example somewhere :)

Comment (soulmonk) [reply .eslintrc.json:119]: this is more than enough )

Comment (hilleer) [reply .eslintrc.json:119]: They seem to be doing something like it in airbnb but their solution do not seem simple :-D

File: package.json [hunk 2]
```diff
@@ -22,8 +22,9 @@
   },
   "homepage": "https://github.com/debitoor/eslint-config-debitoor#readme",
   "devDependencies": {
-    "mocha": "5.2.0",
-    "mocha-eslint": "5.0.0"
+    "mocha": "9.0.1",
+    "mocha-eslint": "7.0.0",
+    "eslint": "7.29.0"
```
Comment (hilleer) [inline package.json:27]: I don't see any addition of tests etc.. Why is this required as a dev dependency? 👁️

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@debitoor/eslint-config-debitoor",
-  "version": "3.1.0",
+  "version": "3.2.0",
```
Comment (hilleer) [inline package.json:3]: 1. I think the ecma version bump might be considered a major (breaking) change
2. use [npm version](https://docs.npmjs.com/cli/v8/commands/npm-version) to bump packages

Comment (soulmonk) [reply package.json:3]: good point )

File: package.json (hunk 2 above)
Comment (soulmonk) [reply package.json:27]: no idea, I've just update them, will remove

**Comment by soulmonk** [comment]
Obsolete.
